client/src/pages/ClientManagementPage/ClientManagementPage.js

```diff
@@ -32,10 +33,34 @@ function ClientManagementPage(props) {
   const [clientInfoList, setClientInfoList] = useState(null);
   const [isClientInfoListLoading, setIsClientInfoListLoading] = useState(false);
   const [requestError, setRequestError] = useState("");
+  const [hasPendingCancelationFilter, setHasPendingCancelationFilter] = useState("All");
+
+  function onCoveredOrderFilterChange(newVal) {
+    setHasPendingCancelationFilter(newVal);
+    onSearchClientButtonClick();
```

Comment:
you can use a use effect like we did in the ProductManagementPage.js in order to follow the same pattern.

Reply:
i decided to not implement the "auto button click". So i deleted it. It makes things easier and it's better